src/components/modal/modal.tsx

```diff
@@ -129,7 +129,7 @@ export class Modal implements ConditionalSlotComponent, OpenCloseComponent {
   //--------------------------------------------------------------------------
   componentWillLoad(): void {
     // when modal initially renders, if active was set we need to open as watcher doesn't fire
-    if (this.open) {
+    if (this.open || this.active) {
```

Comment:
I think this file needs some more work so that there is only one source of truth. Inlcuding...

```
if (this.open || this.active) {
```

Another example:

```ts
@Watch("active")
  @Watch("open")
```

```
 private openModal() {
    this.previousActiveElement = document.activeElement as HTMLElement;
    this.el.addEventListener("calciteModalOpen", this.openEnd);
    this.open = true;
    this.active = true;
```

```
close = (): Promise<void> => {
    return this.beforeClose(this.el).then(() => {
      this.open = false;
      this.active = false;
      focusElement(this.previousActiveElement);
      this.removeOverflowHiddenClass();
    });
  };
```

Reply:
The current code base is already working as one source of truth. Watchers for both `active` & `open` trigger either `openModal( )` or `closeModal( )` which sync up both active and open values.

Reply:
`if (this.open || this.active) {
  this.openModal( )
  }` 
  inside  componentWillLoad( ) will invoke `openModal( )` if either of open or active is true and openModal method sets both active and open to `true`.

Reply:
Ok. I was just thinking it might be better to just have `active` as a property that sets `open` in a watcher. Then `active` doesn't need to be used in the methods or render functions and will be easier to remove when we release 1.0 and can remove deprecated properties.

Reply:
I was thinking of avoiding the additional `watchers` code which is expensive. Agree with the fact it is easier to clean up in the future.